scipy/stats/_distribution_infrastructure.py

```diff
@@ -3582,12 +3610,21 @@ def _overrides(self, method_name):
             return True
         return super()._overrides(method_name)
 
-    # These should probably check whether pmf = 0
     def _logpdf_formula(self, x, **params):
-        return np.full_like(x, np.inf)
+        if params:
+            p = next(iter(params.values()))
+            nan_result = np.isnan(x) | np.isnan(p)
+        else:
+            nan_result = np.isnan(x)
```

Comment:
Another use for `_invalid`. If it doesn't do the right thing, looks like we need something that does. Maybe it would look like:
```suggestion
        nan_result = np.isnan(x)
        if params:
            nan_result = nan_result | self._invalid
```

or just

```suggestion
        nan_result = nan_result | self._invalid
```
for simplicity.